### ISSUE TITLE
Packet.net Machine never comes back from upgrade to 1.0

### ISSUE THREAD

**Issue by withinboredom** [rancher/os#1789]
**RancherOS Version: (ros os version)**
Was 0.9.0, upgrade to 1.0

**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
Running at Packet.

Went through normal upgrade process ... machine never came back online after reboot, no errors in Console (actually nothing at all in console for that matter). I'm able to repeatedly replicate:

1. Start a RancherOS installation on Packet, type-0. 
2. Upgrade to 0.9.0
3. Upgrade to 1.0
4. reboot
4. Never boots...

**Comment by SvenDowideit** [comment]
yup, that's pretty darned weird.

**Comment by SvenDowideit** [comment]
oh, I'm 99% sure this is only because the kernel / RancherOs isn't sending the acpi reboot / restart - if you go to the packet.net web UI, or their cmdline tool and tell it to acutally reboot, it should reboot with 1.0.0 

(tested - got same result, then hit reboot, and it did - but that then shows up the next problem - the network settings need to be changed :/ - I'll have to figure that out for 1.0.1)

basically - it boots, what it's failing to do is tell the hardware to turn off / reset.

**Comment by SvenDowideit** [comment]
running `ros os upgrade -f -i rancher/os:v1.0.1-rc1` appears to have rebooted, but i wasn't watching :/

same "Datasource not ready issue" I'll need to think about for v1.0.1-rc2 - iirc, the kernel boot params need to be changed to use `rancher.network.interfaces.eth*.dhcp=true`

**Comment by SvenDowideit** [comment]
and, yeah, I changed the cmdline after upgrading to `rancher/os:v1.0.1-rc1` and things worked the way we want:

```
[rancher@sven1 ~]$ cat /proc/cmdline
BOOT_IMAGE=../vmlinuz-4.9.22-rancher printk.devkmsg=on rancher.state.dev=LABEL=RANCHER_STATE rancher.state.wait console=tty0 rancher.state.mdadm_scan console=ttyS1,115200n8 rancher.autologin=ttyS1 rancher.cloud_init.datasources=[packet] rancher.network.interfaces.eth*.dhcp=true initrd=../initrd-v1.0.1-rc1
```

(remove the `bond0` parts and replace with one `...dhcp=true`.

to edit the boot line temporarily, hit any key while syslinux is booting, and then type all of it :(

or to edit it permanently, wait for the datasource to timeout, and allow things to finish booting, then run `blkid` to find out what device has the `RANCHER_BOOT` label

```
mount /dev/md124 /mnt
vi /mnt/global
```

edit and reboot.

I'm not yet sure what the safest way to update the boot line is, it might mean that you can't downgrade.

**Comment by withinboredom** [comment]
Thanks for taking a look at this over the weekend! I'm travelling right now, but I will let you know how it goes Monday! I don't think I need to worry about downgrading ;)

**Comment by SvenDowideit** [comment]
so far, my testing suggests that v1.0.1-rc3 (hopefully out in the next 24 hrs) will fix this. I'm just shaking it to see what breaks.

**Comment by withinboredom** [comment]
I forgot to report back! Indeed, everything is fixed!
